frontend/src/static/js/v_authorship.js

```diff
@@ -313,7 +328,8 @@ window.vAuthorship = {
       this.$store.commit('incrementLoadingOverlayCount', 1);
       setTimeout(() => {
         this.selectedFiles = this.files.filter(
-            (file) => this.selectedFileTypes.includes(file.fileType)
+            (file) => ((this.selectedFileTypes.includes(file.fileType) && !file.isBinary)
+            || (file.isBinary && this.isBinaryFilesChecked))
```

Comment:
just wondering why the condition so complicated. selectedfiletypes should not contain any binary file?

Reply:
Selected file types do not include binary files because binary file has 0 line count. I also think this should make sense because strictly speaking binary is not really a file type, rather it should be a file type group, containing file types such as `png` and `jpg`, and therefore it may be necessary to separate binary files with normal files.

Reply:
So what if I specify jpg in file type? If I tick binary but untick jpg? Could you please set up a repo and respective argument for us to test?

Reply:
Thanks for pointing it out. `jpg` and `png` will be grouped under 'binary' if they are specified by the user. If the user specifies `jpg` in the format argument, the user will see the `jpg` files in under `binary`. Therefore, technically speaking the `jpg` checkbox will not appear.

Reply:
So let me check the logic here, the `this.selectedFileTypes` will never include any binary file types according to your explanation? Then what does the condition `this.selectedFileTypes.includes(file.fileType) && !file.isBinary` do?

Reply:
After some trial, I found that this check is necessary because there can be a situation where the user specifies the file group for a repository. In this case, it is possible that a binary file will be included in both the file group and the binary group without this condition.

Reply:
Could you please provide an example for this?

Reply:
![image](https://user-images.githubusercontent.com/54735123/108589336-d0309480-7398-11eb-875e-54ff061da053.png)

There can be issue caused by special name, especially related to the `other` category. When the extension of the binary file cannot be recognized, it will be both in the `other` group and in the binary group without this condition.

Reply:
@Tejas2805 @fzdy1914 any idea for the file type handling here？

Reply:
I think if not recognized, we can just keep it in 'other' and not in 'binary'. This will be consistent with what we mention in the docs regarding which file formats are recognized as `binary`.

Reply:
@fzdy1914 What do you think?

Reply:
I am okay with the current implementation based on what I saw in the deployed report.

Reply:
Uh I think the intention of the original author is that binary file should not be recognized only by its extension but also by the actual file content, because there can be binary file with extension intentionally changed or deleted, and normal file with binary extension. In this case it may not be sufficient to decide whether it is a binary file based on the extension. One example that I can think of is in https://github.com/reposense/testrepo-Alpha.git, specifically in the `My picture` folder and the `binaryFileTest` folder, where whether a file is binary or not should not be decided simply on the extension. This should be part of the reason why the file above without recognizable extension is still recognized as a binary file here, which seems to be correct implementation on my personal opinion. However, I am not sure about this and please let me know if you have opinon on this specific issue.